src/main/java/org/shanerx/tradeshop/admin/AdminEventListener.java

```diff
@@ -40,89 +42,89 @@ public AdminEventListener(TradeShop instance) {
 		plugin = instance;
 	}
 	
-	@EventHandler
+	@EventHandler(priority = EventPriority.HIGH)
 	public void onBlockBreak(BlockBreakEvent event) {
 		Player player = event.getPlayer();
 		Block block = event.getBlock();
 		
 		if (isSign(block)) {
 			Sign s = (Sign) block.getState();
 			
-			if (!isShopSign(s.getBlock()))
+			if (!isShopSign(s.getBlock())) {
 				return;
-			
-			if (player.hasPermission(getAdminPerm()))
+				
+			} else if (player.hasPermission(getAdminPerm())) {
 				return;
-			
-			if (s.getLine(3) == null || s.getLine(3).equals(""))
+				
+			} else if (s.getLine(3) == null || s.getLine(3).equals("")) {
 				return;
-			
-			if (s.getLine(3).equalsIgnoreCase(player.getName()))
+				
+			} else if (getShopOwners(s).contains(Bukkit.getOfflinePlayer(event.getPlayer().getUniqueId()))) {
 				return;
-			
+				
+			}
 			event.setCancelled(true);
 			player.sendMessage(colorize(getPrefix() + plugin.getMessages().getString("no-ts-destroy")));
-			return;
```

Comment:
The method's return type is void and it was the last statement of the if block. After that there is only an else block, so no need for it.